content/en/docs/reference/access-authn-authz/validating-admission-policy.md

```diff
@@ -365,3 +366,122 @@ HTTP response code, are used in the HTTP response to the client.
+However, in the situation where the message expression fails to evaluate, the result will fall back to use the static message instead.
+
+### Type Checking
+A common error when authoring a policy is type confusion. When a policy definition is being created or updated,
+its expressions will be checked against the matched types of `spec.matchConstraints`. The result of type checking
+can be retrieved from `status.typeChecking`. The presence of `status.typeChecking` indicates the completion of type checking.
+An empty `status.typeChecking` means the type checking detects no errors.
+
+For example, given the following policy definition:
+```yaml
+apiVersion: admissionregistration.k8s.io/v1alpha1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "deploy-replica-policy.example.com"
+spec:
+  matchConstraints:
+    resourceRules:
+    - apiGroups:   ["apps"]
+      apiVersions: ["v1"]
+      operations:  ["CREATE", "UPDATE"]
+      resources:   ["deployments"]
+  validations:
+  - expression: "object.replicas > 1" # should be "object.spec.replicas > 1"
+    message: "must be replicated"
+    reason: Invalid
+```
+The status will yield the following information:
+```yaml
+status:
+  typeChecking:
+    expressionWarnings:
+    - fieldRef: spec.validations[0].expression
+      warning: |-
+        apps/v1, Kind=Deployment: ERROR: <input>:1:7: undefined field 'replicas'
+         | object.replicas > 1
+         | ......^
+```
+If multiple resources are matched in `spec.matchConstraints`, all of matched resources will be checked against.
+For example the following policy definition 
+```yaml
+apiVersion: admissionregistration.k8s.io/v1alpha1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "replica-policy.example.com"
+spec:
+  matchConstraints:
+    resourceRules:
+    - apiGroups:   ["apps"]
+      apiVersions: ["v1"]
+      operations:  ["CREATE", "UPDATE"]
+      resources:   ["deployments","replicasets"]
+  validations:
+  - expression: "object.replicas > 1" # should be "object.spec.replicas > 1"
+    message: "must be replicated"
+    reason: Invalid
+```
+will have multiple types in the warning message.
+```yaml
+status:
+  typeChecking:
+    expressionWarnings:
+    - fieldRef: spec.validations[0].expression
+      warning: |-
+        apps/v1, Kind=Deployment: ERROR: <input>:1:7: undefined field 'replicas'
+         | object.replicas > 1
+         | ......^
+        apps/v1, Kind=ReplicaSet: ERROR: <input>:1:7: undefined field 'replicas'
+         | object.replicas > 1
+         | ......^
+```
+
+Type Checking has the following limitation:
+
+- No wildcard matching. If `spec.matchConstraints.resourceRules` contains `"*"` in any of `apiGroups`, `apiVersions` or `resources`,
+  the `"*"` entry will be ignored;
+- The number of matched types is limited to 10. This is to prevent a policy that manually specifying too many types
+  to consume excessive computing resources;
```

Comment:
State what happens when the limit is exceeded?

Reply:
(added)